text/0000-windows-security-identifiers.md

```diff
@@ -0,0 +1,253 @@
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Every platform/implementation-generated image layer entry that currently sets an explicit `tar.Header.Uid/Gid`, will need to instead set a `tar.Header.PAXRecords["MSWINDOWS.rawsd"]` entry based on the SID-user's permissions.
+
+When Docker (and other container runtimes) read an image layer file and create a corresponding container file, the permissions are based on the layer's tar entry headers. For Linux images, the `tar.Header.Uid/Gid` becomes the file's `Uid/Gid`. For Windows, `tar.Header.Uid/Gid` are ignored and instead `tar.Header.PAXRecords["MSWINDOWS.rawsd"]` becomes the file's Security Descriptor (when blank, a permissive default is used).
+
+The value of `MSWINDOWS.rawsd` is a base64-encoded, binary Security Descriptor containing an owner SID and group SID. The Security Descriptor data structure is complex to generate and requires some background.
+
+### Generating Security Descriptors
+Windows Security Descriptor-based file permissions are much richer and more complex than POSIX UID/GIDs, still consisting an owner and group, but whose values can be SIDs of users/groups/other Security Principals (plus optional Access Control Lists, for fine-grained access control, but which can be left blank for our purposes). 
+
+Windows users typically manipulate Security Descriptors through GUIs or complex hierarchical permission systems like Active Directory. Due to this, the data structures to store SIDs themselves and APIs to manipulate them are more complex, less user-friendly, and unique to Windows. 
+
+For example, the Golang Windows-equivalent of `os.Chown`(which [has no Windows implemention](https://golang.org/pkg/os/#Chown)) looks like this:
+  
+```go
+// convert SID string to struct
+ownerSID, err := windows.StringToSid("S-1-5-93-2-1")
+// ... 
+
+// write a owner/group SIDs to the file's security descriptor
+err = windows.SetNamedSecurityInfo(
+    filePath,                           // path to file on filesystem
+    windows.SE_FILE_OBJECT,             // type of object (file)
+    windows.OWNER_SECURITY_INFORMATION, // fields of the security descriptor to write (owner only)
+    ownerSID,                           // owner SID
+    nil,                                // group SID
+    nil,                                // system ACL
+    nil,                                // discretionary ACL
+  )
+// ... 
+```
+
+
+The `StringToSid` function converts the string SID to a struct and `SetNamedSecurityInfo` changes the Owner field of the file's on-disk Security Descriptor. The `nil` fields - Group, DACL, SACL - are not set here but can be set in this API. 
+
+Unfortunately these functions can't be used in any cross-platform way, as both `windows.StringToSid` and `windows.SetNamedSecurityInfo` wrap underlying Windows-only API syscalls.
```

Comment:
Maybe it's worth calling out here why this is problematic - e.g., pack on Linux/Darwin building Windows builders / buildpackages I believe?

Reply:
Great call. Now updated